lib/TestReportHandler.rb

```diff
@@ -127,11 +127,23 @@ def pass
     return 
   end 
 
+#	Might need to alter ‘score’ calculation based on including/not including setup or teardown failures
```

Comment:
```suggestion
# Might need to alter ‘score’ calculation based on including/not including setup or teardown failures
```

Reply:
I agree that it is better to calculate while the TestReport is running, but to my knowledge it should alter existing methods to do so. For example, within the method 'error' it is agnostic of context whether it is setup, test, or teardown. That was the reason why I put the calculation at the end of running. Any idea?

  def error message
    current_action(message).result.code = 'error'
    @num_failures += 1
    return
  end

Reply:
So I tried to code your idea, and it's quite of change in the engine (or maybe I'm missing simple way to do it) To count failure while TS is being run, the code should be in 'fail' in TestReportHandler, for example

  def fail message_type
    @num_failures_testonly +=1
   return 

But to know the context that it is 'test' running (not setup or teardown), the information has to be captured within the method, for example:

  def fail message_type, script
    @num_failures_testonly +=1 if script.test (this may be wrong code but I put it that way)
   return 

Do you have any idea?

Reply:
Okay, let's not worry about it for now then. I have a feeling the ReportHandler is gonna have some pretty significant clean-up done to it anyways, so probably best not to invest anymore time into patching it up.